minkindr/include/kindr/minimal/implementation/transform-2d-inl.h

```diff
@@ -0,0 +1,145 @@
+#include <limits>
+
+#include <glog/logging.h>
+
+namespace kindr {
+namespace minimal {
+
+template <typename Scalar>
+Transformation2DTemplate<Scalar>::Transformation2DTemplate() {
+  setIdentity();
+}
+
+template <typename Scalar>
+Transformation2DTemplate<Scalar>::Transformation2DTemplate(
+    const Rotation& r_A_B, const Position& A_t_A_B)
+    : r_A_B_(r_A_B), A_t_A_B_(A_t_A_B) {}
+
+template <typename Scalar>
+Transformation2DTemplate<Scalar>::Transformation2DTemplate(
+    const Position& A_t_A_B, const Rotation& r_A_B)
+    : Transformation2DTemplate<Scalar>(r_A_B, A_t_A_B) {}
+
+template <typename Scalar>
+Transformation2DTemplate<Scalar>::Transformation2DTemplate(
+    const TransformationMatrix& T)
+    : Transformation2DTemplate<Scalar>(
+          Rotation2D(T.template topLeftCorner<2, 2>().eval()),
+          T.template topRightCorner<2, 1>().eval()) {
+  constexpr Scalar kScalarOne = static_cast<Scalar>(1.0);
+  constexpr Scalar kEpsilon = std::numeric_limits<Scalar>::epsilon();
+  CHECK_LE((T(2, 2) - kScalarOne), kEpsilon);
+  const Eigen::Matrix<Scalar, 2, 2> rotation_matrix =
+      T.template topLeftCorner<2, 2>().eval();
+  CHECK_LE(rotation_matrix.determinant() - kScalarOne, kEpsilon);
```

Comment:
rotation matrices always have a determinant of `+1`: https://en.wikipedia.org/wiki/Rotation_matrix

Reply:
Yes, but you're checking here if it's a valid rotation matrix so you don't know that yet :) Unless the `TransformationMatrix` guarantees that in which case you can probably also get rid of this check

Reply:
@eggerk I think I don't get your point. `TransformationMatrix` is just a `Eigen::Matrix<Scalar, 3, 3>` matrix, so it will not fail if anything is wrong with the transformation. The goal of the check is to see if the determinant of the rotation matrix is `+1`, or is there something wrong with it?

Reply:
But couldn't the determinant be negative or less than 1, in which case your check would still pass despite the input not being a proper rotation matrix?

Reply:
Hmm, I guess it would rather evaluate to `-2` if the determinant is `-1`, doesn't it?

Reply:
@fabianbl , I agree with @eggerk here. If you want to check proximity to 1 you must make sure that the distance to one is small. But that you don't do. The distance of `x` to `1` you should compute with  `abs(x - 1)`. ( `x-1`  is only the relative displacement and a signed quantity).

Reply:
What about using `CHECK_NEAR`  ? That would make this easier:

```c++
CHECK_NEAR(rotation_matrix.determinant(), kScalarOne, kEpsilon);
```

Reply:
Thanks, now I get it :+1: I was rather thinking about determinants of `-1`, will add `abs(...)`